src/Subject.h

```diff
@@ -7,11 +7,11 @@ class Subject {
     Subject();
     void registerObserver(Observer*); 
     void unregisterObserver();
-    virtual double getValue() const;
+    virtual double getValue()const=0;
 
-private:
+protected:
     void _notifyObserver();
     Observer* mObserver;
-    virtual void setVal(const int val);
+    virtual void setValue(const int val)=0;
```

Comment:
What happens in case value is not integer?

Reply:
By the way that the code is programmed and desingned, value is always integer. The value is a analog read from arduino's analog pins and goes from 0 to 1023. In the case u want a generic Subject we must make Observer generic.